tests/components/nest/sensor_sdm_test.py

```diff
@@ -0,0 +1,248 @@
+        """Capture the callback set by Home Assistant."""
+        self._callback = callback
+
+    async def start_async(self) -> DeviceManager:
+        """Return the fake device manager."""
+        return self._device_manager
+
+    @property
+    async def async_device_manager(self) -> DeviceManager:
+        """Return the fake device manager."""
+        return self._device_manager
+
+    def stop_async(self):
+        """No-op to stop the subscriber."""
+        return None
+
+    def receive_event(self, event_message: EventMessage):
+        """Simulate a received pubsub message, invoked by tests."""
+        # Update device state, then invoke HomeAssistant to refresh
+        self._device_manager.handle_event(event_message)
+        self._callback.handle_event(event_message)
+
+
+async def setup_sensor(hass, devices={}, structures={}):
+    """Set up the platform and prerequisites."""
+    MockConfigEntry(domain=DOMAIN, data=CONFIG_ENTRY_DATA).add_to_hass(hass)
+    device_manager = FakeDeviceManager(devices=devices, structures=structures)
+    subscriber = FakeSubscriber(device_manager)
+    with patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation"
+    ), patch("homeassistant.components.nest.PLATFORMS", [PLATFORM]), patch(
+        "homeassistant.components.nest.GoogleNestSubscriber", return_value=subscriber
+    ):
+        assert await async_setup_component(hass, DOMAIN, CONFIG)
+    await hass.async_block_till_done()
+    return subscriber
+
+
+async def test_thermostat_device(hass):
+    """Test a thermostat with temperature and humidity sensors."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {
+                    "sdm.devices.traits.Info": {
+                        "customName": "My Sensor",
+                    },
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.1,
+                    },
+                    "sdm.devices.traits.Humidity": {
+                        "ambientHumidityPercent": 35.0,
+                    },
+                },
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is not None
+    assert temperature.state == "25.1"
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is not None
+    assert humidity.state == "35.0"
+
+
+async def test_no_devices(hass):
+    """Test no devices returned by the api."""
+    await setup_sensor(hass)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is None
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)